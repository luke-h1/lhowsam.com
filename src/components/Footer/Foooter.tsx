/* eslint-disable react/prop-types */
import React from 'react';
import { FiGithub } from 'react-icons/fi';
import { FiMail } from 'react-icons/fi';
import { FiLinkedin } from 'react-icons/fi';
import {
  FooterContainer,
  FooterWrap,
  SocialMedia,
  SocialMediaWrap,
  SocialIconLink,
  SocialIcons,
  SocialLogo,
} from './FooterElements';


const Footer: React.FC<{theme: String}> = (props) => {
  const theme = props;

  return (
      <FooterContainer>
        <FooterWrap>
          <SocialMedia>
            <SocialMediaWrap>
              <SocialIcons>
                <SocialIconLink
                  href="https://github.com/luke-h1"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <FiGithub />
                </SocialIconLink>
                <SocialIconLink
                  href="https://www.linkedin.com/in/lukehowsam/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <FiLinkedin />
                </SocialIconLink>
                <SocialIconLink
                  href="mailto:luke.howsam@yahoo.com?subject=Get In Touch"
                  aria-label="Email"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <FiMail />
                </SocialIconLink>
              </SocialIcons>
            </SocialMediaWrap>
          </SocialMedia>
        </FooterWrap>
      </FooterContainer>
  );
};

export default Footer;
