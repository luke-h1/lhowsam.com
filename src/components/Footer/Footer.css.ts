import { variables } from '@frontend/styles/variables.css';
import { responsiveStyle } from '@frontend/utils/style.css';
import { style } from '@vanilla-extract/css';

export const root = style({
  position: 'relative',
  margin: '0 auto',
  marginTop: '1rem',
  paddingTop: variables.spacing.xxl,
  paddingBottom: variables.spacing.xxl,
  paddingRight: variables.spacing.md,
  paddingLeft: variables.spacing.md,
  backgroundColor: variables.color.page,
});

export const container = style({
  margin: '0 auto',
  display: 'grid',
  rowGap: variables.spacing.lg,
  ...responsiveStyle({
    sm: {
      gridTemplateColumns: 'repeat(2, 1fr)',
    },
    md: {
      gridTemplateColumns: 'repeat(3, 1fr)',
    },
  }),
});

export const links = style({
  display: 'grid',
  rowGap: variables.spacing.sm,
  color: variables.color.foregroundNeutral,
});

export const copy = style({
  gridColumn: '1 / -1',
  paddingTop: variables.spacing.lg,
  borderTop: `1px solid ${variables.color.border}`,
});
