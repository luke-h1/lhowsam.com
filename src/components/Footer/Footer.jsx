/* eslint-disable react/prop-types */
import React from 'react';
import { SiGithub } from 'react-icons/si';
import { AiOutlineMail } from 'react-icons/ai';
import { ImLinkedin2 } from 'react-icons/im';
import { ThemeProvider } from 'theme-ui';
import {
  FooterContainer,
  FooterWrap,
  SocialMedia,
  SocialMediaWrap,
  SocialIconLink,
  SocialIcons,
  SocialLogo,
} from './FooterElements';
import { CodeIcon } from '../../helpers/Icon/CodeIcon';

const Footer = ({ theme }) => (
  <ThemeProvider theme={theme}>
    <FooterContainer>
      <FooterWrap>
        <SocialMedia>
          <SocialMediaWrap>
            <SocialLogo to="/">
              <CodeIcon />
            </SocialLogo>
            <SocialIcons>
              <SocialIconLink
                href="https://github.com/luke-h1"
                target="_blank"
                rel="noopener noreferrer"
              >
                <SiGithub />
              </SocialIconLink>
              <SocialIconLink
                href="https://www.linkedin.com/in/lukehowsam/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <ImLinkedin2 />
              </SocialIconLink>
              <SocialIconLink
                href="mailto:luke.howsam@yahoo.com?subject=Get In Touch"
                aria-label="Email"
                target="_blank"
                rel="noopener noreferrer"
              >
                <AiOutlineMail />
              </SocialIconLink>
            </SocialIcons>
          </SocialMediaWrap>
        </SocialMedia>
      </FooterWrap>
    </FooterContainer>
  </ThemeProvider>
);
export default Footer;
