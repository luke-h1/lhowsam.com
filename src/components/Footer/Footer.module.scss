@import '../../styles/partials/_mixins.scss';
@import '../../styles/partials/_utils.scss';
@import '../../styles/variables.scss';

.contentFooter {
  display: flex;
  flex-direction: column;

  @include media(sm) {
    flex-direction: row;
  }
}

.presentation {
  width: 100%;
  display: inline-flex;
  flex-direction: column;

  h3 {
    text-align: center;
    font-size: 26px;
    line-height: 2;
    margin-bottom: 8px;
    &:hover {
      cursor: pointer;
    }
  }

  @include media(sm) {
    flex-direction: column;
    width: 40%;

    h3 {
      text-align: left;
    }

    ul {
      justify-content: flex-start;
    }
  }

  @include media(md) {
    width: 50%;
  }
}

.menuFooter {
  width: 100%;
  display: inline-flex;
  flex-direction: row;

  ul {
    margin-top: 32px;
    list-style-type: none;
    flex: 1;

    li {
      display: inline-flex;
      justify-content: center;
      flex-direction: row;
      margin-top: 16px;
      flex-wrap: wrap;
      width: 50%;

      a {
        color: $primary;
        opacity: 0.72;

        &:hover {
          color: $pink;
          opacity: 1;
        }
      }
    }
  }

  @include media(sm) {
    flex-direction: column;
    width: 60%;

    ul {
      margin-top: 0;

      li {
        justify-content: flex-start;
      }
    }
  }

  @include media(md) {
    width: 50%;
  }
}

.aboutFooter {
  text-align: center;
  position: relative;
  color: $primary;
  font-size: 15px;
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: rgba($dark, 0.4);
  padding-top: 32px;
  margin-top: 32px;
}
