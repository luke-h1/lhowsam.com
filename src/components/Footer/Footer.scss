.footer-container {
  background-color: #000;
  padding: 4rem 0 2rem 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 40vh;
  min-width: 100vw;
  color: #fff;
  border-top: 1px solid #fff;
}

.footer-subscription {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin-bottom: 24px;
  padding: 24px;
  color: #fff;
}

.footer-subscription-heading {
  margin-bottom: 24px;
  font-size: 24px;
}

.footer-subscription-text {
  margin-bottom: 24px;
  font-size: 20px;
}

.footer-input {
  padding: 8px 24px;
  border-radius: 2px;
  margin-right: 10px;
  outline: none;
  border: none;
  font-size: 18px;
  margin-bottom: 16px;
  border: 1px solid #fff;
  ::placeholder {
    color: #b1b1b1;
  }
}

.footer-links {
  width: 100%;
  max-width: 1000px;
  display: flex;
  justify-content: center;
}

.footer-link-wrapper {
  display: flex;
}

.footer-link-items {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 16px;
  text-align: left;
  width: 160px;
  box-sizing: border-box;
  h2 {
    margin-bottom: 16px;
  }
  > h2 {
    color: #fff;
  }

  a {
    color: #fff;
    text-decoration: none;
    margin-bottom: 0.5rem;
    &:hover {
      color: #e9e9e9;
      transition: 0.3s ease-out;
    }
  }
}

/* Social Icons */
.social-icon-link {
  color: #fff;
  font-size: 24px;
  cursor: pointer;
  &:hover {
    color: #2ab7ca;
    cursor: pointer;
  }
  a {
    &:hover {
      color: #2ab7ca;
      cursor: pointer;
    }
  }
}

.social-media {
  max-width: 1000px;
  width: 100%;
}

.social-media-wrap {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  max-width: 1000px;
  margin: 40px auto 0 auto;
  text-align: center;
}

.social-icons {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100px;
  cursor: pointer;
}

.social-logo {
  color: #fff;
  justify-self: start;
  margin-left: 20px;
  cursor: pointer;
  text-decoration: none;
  font-size: 2rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 20px;
}

.website-rights {
  color: #fff;
  margin-bottom: 15px;
  text-align: center;
}

@media (max-width: 820px) {
  .footer-links {
    padding-top: 2rem;
  }

  .footer-input {
    width: 100%;
  }

  .btn {
    width: 100%;
  }

  .footer-link-wrapper {
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .social-media-wrap {
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
}

@media (max-width: 640px) {
  .footer-icon {
    margin-right: 20px;
  }
}
