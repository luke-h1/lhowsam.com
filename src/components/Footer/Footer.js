import React from 'react';
import './Footer.scss';
import { Link } from 'react-router-dom';
import { FaGithub, FaLinkedin } from 'react-icons/fa';
import { FiDroplet } from 'react-icons/fi';
import PropTypes from 'prop-types';
const Footer = ({copyright }) => {
  return (
    <div className="footer-container">
      <section className="social-media">
        <div className="social-media-wrap">
          <div className="footer-logo">
            <Link to="/" className="social-logo">
              <FiDroplet className="navbar-icon" />
            </Link>
          </div>
          <small className="website-rights">
            {copyright} {new Date().getFullYear()}
          </small>
          <div className="social-icons">
            <a
              className="social-icon-link"
              href="https://github.com/luke-h1"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Github"
            >
              <FaGithub test-id="github" />
            </a>
            <a
              className="social-icon-link"
              href="https://www.linkedin.com/in/lukehowsam/"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="LinkedIn"
            >
              <FaLinkedin test-id="linkedin" />
            </a>
          </div>
        </div>
      </section>
    </div>
  );
};

Footer.defaultProps = {
  copyright: 'Luke Howsam Â©',
};

Footer.propTypes = {
  copyright: PropTypes.string.isRequired,
};

export default Footer;
