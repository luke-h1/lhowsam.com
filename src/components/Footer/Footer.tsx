import dynamic from 'next/dynamic';
import Box from '../Box/Box';
import Heading from '../Heading/Heading';
import Link from '../Link/Link';
import Spacer from '../Spacer/Spacer';
import Text from '../Text/Text';
import * as styles from './Footer.css';
import ThemeSelect from './ThemeSelect/ThemeSelect';

const NowPlaying = dynamic(() => import('../NowPlaying/NowPlaying'), {
  ssr: true,
});

interface FooterLink {
  id: number;
  text: string;
  href: string;
}

const links: FooterLink[] = [
  {
    id: 1,
    text: 'Home',
    href: '/',
  },
  {
    id: 2,
    text: 'About',
    href: '/about',
  },
  {
    id: 3,
    text: 'Blog',
    href: '/blog',
  },
  {
    id: 4,
    text: 'Projects',
    href: '/projects',
  },
];

const socialLinks: FooterLink[] = [
  {
    id: 1,
    text: 'Github',
    href: 'https://github.com/luke-h1',
  },
  {
    id: 2,
    text: 'Linkedin',
    href: 'https://www.linkedin.com/in/lukehowsam',
  },
  {
    id: 3,
    text: 'Email',
    href: 'mailto:luke.howsam@yahoo.com',
  },
];

const Footer = () => {
  return (
    <Box
      as="article"
      className={styles.root}
      maxWidth={{ md: 'text' }}
      data-testid="footer"
    >
      <div className={styles.container}>
        <div>
          <Heading color="foregroundNeutral">Navigate</Heading>
          <Spacer height="md" />
          <ul className={styles.links}>
            {links &&
              links.map(link => (
                <li key={link.id}>
                  <Link href={link.href}>{link.text}</Link>
                </li>
              ))}
          </ul>
        </div>
        <div>
          <Heading color="foregroundNeutral">Get in touch</Heading>
          <Spacer height="md" />
          <ul className={styles.links}>
            {socialLinks &&
              socialLinks.map(link => (
                <li key={link.id}>
                  <Link href={link.href}>{link.text}</Link>
                </li>
              ))}
          </ul>
        </div>
        <div>
          <ThemeSelect />
        </div>
        <div className={styles.copy}>
          <Text color="foregroundNeutral" fontSize="sm">
            <NowPlaying />
            {new Date().getFullYear()} - lhowsam.com
          </Text>
        </div>
      </div>
    </Box>
  );
};

export default Footer;
