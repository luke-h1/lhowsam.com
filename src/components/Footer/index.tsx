import { Heading } from '../Heading';
import Link from '../Link';
import { List } from '../List';
import NowPlaying from '../NowPlaying';
import { Spacer } from '../Spacer';
import Text from '../Text';
import ThemeSelect from '../ThemeSelect';
import * as styles from './Footer.css';

interface FooterLink {
  id: number;
  text: string;
  href: string;
}

const navigationLinks: FooterLink[] = [
  {
    id: 1,
    text: 'Home',
    href: '/',
  },
  {
    id: 2,
    text: 'About',
    href: '/about',
  },
  // {
  //   id: 3,
  //   text: 'Experience',
  //   href: '/experience',
  // },
  {
    id: 4,
    text: 'Blog',
    href: '/blog',
  },
  {
    id: 5,
    text: 'Projects',
    href: '/projects',
  },
  {
    id: 6,
    text: 'Talks',
    href: '/talks',
  },
];

const socialLinks: FooterLink[] = [
  {
    id: 1,
    text: 'Github',
    href: 'https://github.com/luke-h1',
  },
  {
    id: 2,
    text: 'Linkedin',
    href: 'https://www.linkedin.com/in/lukehowsam',
  },
  {
    id: 3,
    text: 'Email',
    href: 'mailto:luke.howsam@yahoo.com?subject=Get in touch (lhowsam.com)',
  },
];

const Footer = () => {
  return (
    <footer className={styles.root} data-testid="footer">
      <div className={styles.container}>
        <div>
          <Heading>Navigate</Heading>
          <Spacer height="md" />
          <List className={styles.links} data-testid="footer-navigation-links">
            {navigationLinks &&
              navigationLinks.map(link => (
                <li key={link.id}>
                  <Link href={link.href}>{link.text}</Link>
                </li>
              ))}
          </List>
        </div>
        <div>
          <Heading>Social</Heading>
          <Spacer height="md" />
          <List className={styles.links} data-testid="footer-social-links">
            {socialLinks &&
              socialLinks.map(link => (
                <li key={link.id}>
                  <Link href={link.href}>{link.text}</Link>
                </li>
              ))}
          </List>
        </div>
        <div>
          <ThemeSelect />
        </div>
        <div className={styles.note} data-testid="footer-cc">
          <Text color="foregroundNeutral" fontSize="sm" fontFamily="mono">
            Â© {new Date().getFullYear()} Luke Howsam
          </Text>
        </div>
        <div className={styles.copy}>
          <NowPlaying />
        </div>
      </div>
    </footer>
  );
};

export default Footer;
