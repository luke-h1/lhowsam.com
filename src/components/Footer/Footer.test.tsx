/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';
import { theme } from '../../styles/Themes';


test('Footer matches snapshot', () => {
    // @ts-ignore
  const footer = renderer.create(<Footer theme={theme}/>)


    .toJSON();
  expect(footer).toMatchSnapshot();
});