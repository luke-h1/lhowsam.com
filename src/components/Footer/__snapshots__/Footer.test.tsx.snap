// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer
  className="sc-bdfBwQ hRrhXX"
>
  <div
    className="sc-gsTCUz xLEuc"
  >
    <section
      className="sc-dlfnbm ixTiAE"
    >
      <div
        className="sc-hKgILt eyfALI"
      >
        <a
          className="sc-eCssSg hYWkeK"
          href="/"
        >
          <svg
            className="sc-iBPRYJ lfpeRP"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20,3H4C2.897,3,2,3.897,2,5v14c0,1.103,0.897,2,2,2h16c1.103,0,2-0.897,2-2V5C22,3.897,21.103,3,20,3z M4,19V7h16 l0.002,12H4z"
            />
            <path
              d="M9.293 9.293L5.586 13 9.293 16.707 10.707 15.293 8.414 13 10.707 10.707zM14.707 9.293L13.293 10.707 15.586 13 13.293 15.293 14.707 16.707 18.414 13z"
            />
          </svg>
        </a>
        <div
          className="sc-jSgupP jgQhSU"
        >
          <a
            className="sc-gKsewC laZCvl"
            href="https://github.com/luke-h1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title />
              <path
                d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
              />
            </svg>
          </a>
          <a
            className="sc-gKsewC laZCvl"
            href="https://www.linkedin.com/in/lukehowsam/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 6h2.767v1.418h0.040c0.385-0.691 1.327-1.418 2.732-1.418 2.921 0 3.461 1.818 3.461 4.183v4.817h-2.885v-4.27c0-1.018-0.021-2.329-1.5-2.329-1.502 0-1.732 1.109-1.732 2.255v4.344h-2.883v-9z"
              />
              <path
                d="M1 6h3v9h-3v-9z"
              />
              <path
                d="M4 3.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5c0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5z"
              />
            </svg>
          </a>
          <a
            aria-label="Email"
            className="sc-gKsewC laZCvl"
            href="mailto:luke.howsam@yahoo.com?subject=Get In Touch"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
              />
            </svg>
          </a>
        </div>
      </div>
    </section>
  </div>
</footer>
`;
